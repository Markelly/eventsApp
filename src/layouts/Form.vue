<template>
  <div class="form">
    <h1 v-text="title"></h1>
    <b-form class="form__fields" @submit.prevent="submit">
      <slot name="form-fields"></slot>
      <b-button
        class="form__button"
        type="submit"
        variant="primary"
        v-text="buttonLabel"
      />
    </b-form>
    <a class="form-link" v-text="footerLabel" @click="goToPage"></a>
  </div>
</template>

<script>
export default {
  name: 'Form',
  props: {
    title: {
      type: String,
      default: ''
    },
    buttonLabel: {
      type: String,
      default: ''
    },
    footerLabel: {
      type: String,
      default: ''
    },
    footerLink: {
      type: String,
      default: ''
    }
  },
  methods: {
    submit() {
      this.$emit('submit')
    },
    goToPage(){
      this.$router.push(this.footerLink);
    }
  }
}
</script>

<style scoped lang="scss">
.form {
  max-width: 480px;
  margin: 80px auto;

  &__fields {
    margin: 16px 0;
  }

  &__button {
    margin-top: 8px;
  }
}
</style>
