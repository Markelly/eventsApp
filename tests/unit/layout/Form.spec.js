import { createLocalVue, shallowMount } from '@vue/test-utils'
import { BootstrapVue } from 'bootstrap-vue'
import Form from '@/layouts/Form'

const localVue = createLocalVue()
localVue.use(BootstrapVue)

describe('<Form />', () => {
  let wrapper

  beforeEach(() => {
    wrapper = shallowMount(Form, {
      localVue
    })
  })

  test('should render the component', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should emit an event to submit form', () => {
    wrapper.vm.submit()
    expect(wrapper.emitted('submit')).toBeTruthy()
  })
})
