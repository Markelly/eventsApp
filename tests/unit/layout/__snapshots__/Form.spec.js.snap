// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Form /> should render the component 1`] = `
<div class="form">
  <h1></h1>
  <b-form-stub class="form__fields">
    <b-button-stub variant="primary" type="submit" tag="button" target="_self" class="form__button"></b-button-stub>
  </b-form-stub> <a class="form-link"></a>
</div>
`;
